src/ScottLogic.Internal.Training/ScottLogic.Internal.Training.Api/Controllers/UsersController.cs

```diff
@@ -98,20 +99,57 @@ public async Task<IActionResult> AddUser([FromBody] User user)
         [ProducesResponseType(StatusCodes.Status200OK)]
         [ProducesResponseType(StatusCodes.Status400BadRequest)]
         [ProducesResponseType(StatusCodes.Status404NotFound)]
-        [Authorize(Roles = "Administrator")]
+        [Authorize]
         [HttpDelete]
         public async Task<IActionResult> Delete([FromBody] User user)
         {
-            var users = await _context.Users.ToArrayAsync();
             if (user.Username != null)
             {
-                if (users.Any(currentUser => currentUser.Username == user.Username))
+                // Get the user Role from the access token
+                var userIdentity = this.User.Identity as ClaimsIdentity;
+                var userRole = UserRole.None;
+                if (userIdentity != null && userIdentity.HasClaim(c => c.Type == ClaimTypes.Role))
                 {
-                    _context.Users.Remove(_context.Users.FirstOrDefault(u => u.Username == user.Username));
-                    _context.SaveChanges();
-                    return Ok("User removed from the database");
+                    var userRolesString = userIdentity.Claims
+                        .FirstOrDefault(c => c.Type == ClaimTypes.Role).Value;
+                    if(userRolesString == "Admin")
+                    {
+                        userRole = UserRole.Admin;
+                    }
+                }
+                // Get the existing users from the database
+                var users = await _context.Users.ToArrayAsync();
+                
+                // Delete the user passed in the request
+                if(userRole == UserRole.Admin)
+                {
+                    if (users.Any(currentUser => currentUser.Username == user.Username))
```

Comment:
This bit of code is repeated below.

Can we refactor to extract the name out before we do this. We can then do something like:

if (userRole == UserRole.Admin || users.Any(currentUser => currentUser.Username == userName)){
// delete code
}